src/main/scala/mesosphere/marathon/storage/repository/HealthCheckShieldRepository.scala

```diff
@@ -19,17 +19,33 @@ import mesosphere.marathon.core.storage.store.{IdResolver, PersistenceStore}
 import mesosphere.marathon.state.Timestamp
 
 trait HealthCheckShieldRepository
-  extends Repository[Task.Id, HealthCheckShield]
+  extends Repository[HealthCheckShield.Id, HealthCheckShield]
 
 object HealthCheckShieldRepository {
   implicit val category = "HealthCheckShield"
 
+  val v1ShieldName = "v1_shield"
+  val v2StoragePrefix = "v2_"
+
   implicit val byteOrder: ByteOrder = ByteOrder.BIG_ENDIAN
 
-  implicit val zkIdResolver = new IdResolver[Task.Id, HealthCheckShield, String, ZkId] {
-    def toStorageId(id: Task.Id, version: Option[OffsetDateTime]): ZkId =
-      ZkId(category, id.idString, version)
-    def fromStorageId(key: ZkId): Task.Id = Task.Id.parse(key.id)
+  implicit val zkIdResolver = new IdResolver[HealthCheckShield.Id, HealthCheckShield, String, ZkId] {
```

Comment:
During live review, I thought that in case of revert we could simply remove the shield from zookeeper, but I don't think it's serialized in its own path; it is serialized along with apps... So in case of rollback, we may need to wait for it to expire before so that v2 shields don't make marathon crash.

Reply:
They should propagate `category=HealthCheckShield` to the zookeeper key, so it should be on its own path.

Actually they have this code in `ZkPersistenceStore`:
```
case class ZkId(category: String, id: String, version: Option[OffsetDateTime]) {
  private val bucket = math.abs(id.hashCode % ZkId.HashBucketSize)

  // BUG: id = "" for the root group this results in "Path must not end with / character" in curator
  def path: String = version.fold(f"/$category/$bucket%x/$id") { v =>
    val truncatedVersion = v.truncatedTo(ChronoUnit.MILLIS)
    f"/$category/$bucket%x/$id/${ZkId.DateFormat.format(truncatedVersion)}"
  }
}
```
However I'm a bit confused with the bucket part, and I also see I need to validate `shieldName` to not be empty...

Reply:
Oh you're right; I don't know why but there was nothing on am6 preprod but indeed there is a dedicated path in par:
```
[zk: localhost:2181(CONNECTED) 11] ls /marathon/state/HealthCheckShield
[1, 2, 7]
```
So it should globally be fine I guess :)